File: api/src/main/java/dev/ulloasp/mlsuite/organization/application/dto/OrganizationAdminDashboardDto.java
package dev.ulloasp.mlsuite.organization.application.dto;

import java.util.List;

import dev.ulloasp.mlsuite.invitation.application.dto.InvitationDto;
import dev.ulloasp.mlsuite.team.application.dto.TeamDto;
import dev.ulloasp.mlsuite.workspace.application.dto.WorkspacePermissionsDto;

public record OrganizationAdminDashboardDto(
        OrganizationDto organization,
        WorkspacePermissionsDto permissions,
        OrganizationAdminStatsDto stats,
        List<TeamDto> recentTeams,
        List<OrganizationMembershipRowDto> recentMembers,
        List<InvitationDto> recentInvitations) {
}

File: frontend/src/api/workspace/dtos/organization-admin-dashboard-dto.ts
/*
SPDX-License-Identifier: MIT
Copyright (c) 2025 Pablo Ulloa Santin
*/

import type {
  WorkspacePermissionsDto,
  OrganizationDto,
  OrganizationMembershipRowDto,
  TeamDto,
  InvitationDto,
  OrganizationAdminStatsDto,
} from "./index";

export interface OrganizationAdminDashboardDto {
  organization: OrganizationDto;
  permissions: WorkspacePermissionsDto;
  stats: OrganizationAdminStatsDto;
  recentTeams: TeamDto[];
  recentMembers: OrganizationMembershipRowDto[];
  recentInvitations: InvitationDto[];
}
